### PR TITLE
Attempt to close #293

### LINKED ISSUE #293: precompilation warning
On `1.9.0-rc2`:
```julia
WARNING: Method definition init_cacheval(LinearSolve.GenericFactorization{F} where F, Any, Any, Any, Any, Any, Int64, Any, Any, Bool, LinearSolve.OperatorAssumptions{issq, condition} where condition where issq) in module LinearSolve at [...]/.julia/packages/LinearSolve/P5a5J/src/factorization.jl:313 overwritten at [...]/.julia/packages/LinearSolve/P5a5J/src/factorization.jl:326.
  ** incremental compilation may be fatally broken for this module **
```

On `[7ed4a6bd] LinearSolve v1.41.1`.

### PR DESCRIPTION
**Pull request by thomvet**
- Simplified dispatch in factorization.jl
- Tests pass locally
- Also brought back cholesky! tests (needed a "don't alias" fix in basictests.jl)

### CODE REVIEW

File: src/factorization.jl [hunk 1]
```diff
@@ -310,21 +310,20 @@ end
...
+#                        A, b, u, Pl, Pr, maxiters::Int, abstol, reltol, verbose::Bool,
+#                        assumptions::OperatorAssumptions)
+#     newA = copy(convert(AbstractMatrix, A))
+#     if newA isa AbstractSparseMatrix
+#         fill!(nonzeros(newA), true)
+#     else
+#         fill!(newA, true)
+#     end
+#     do_factorization(alg, newA, b, u)
+# end
```
Comment (ChrisRackauckas) [inline src/factorization.jl:323]:
```suggestion
```

File: test/basictests.jl [hunk 2]
```diff
@@ -39,6 +39,7 @@ function test_interface(alg, prob1, prob2)
     sol = solve(cache; cache_kwargs...)
     @test A2 * sol.u ≈ b1
 
+    cache = LinearSolve.set_A(cache, deepcopy(A2))
```
Comment (ChrisRackauckas) [inline test/basictests.jl:42]: this is no longer testing that the no-aliasing works.

Comment (thomvet) [reply test/basictests.jl:42]: I added that to make the last test work for `cholesky!`. Does that indicate that the no-aliasing behavior is not correctly defined for this algorithm? Where?

Comment (ChrisRackauckas) [reply test/basictests.jl:42]: Seems like it. `default_alias_A(::Any, ::Any, ::Any) = false` should make it internally deepcopy. This this is a separate topic so it shouldn't be in the same PR.

**Review by ChrisRackauckas — APPROVED** [review]

File: test/basictests.jl (hunk 2 above)
Comment (thomvet) [reply test/basictests.jl]: Yeah, I saw that and thought that it should do the deep copy internally, but the failing test tells me that it isn't *shrug*. I reverted back to not deep-copying here and re-commented the failing test.

**Pull request merged**